homeassistant/helpers/entity_registry.py

```diff
@@ -289,7 +289,7 @@ def async_generate_entity_id(
         if len(domain) > MAX_LENGTH_STATE_DOMAIN:
             raise MaxLengthExceeded(domain, "domain", MAX_LENGTH_STATE_DOMAIN)
 
-        test_string = preferred_string
+        test_string = preferred_string[:MAX_LENGTH_STATE_ENTITY_ID]
         if not known_object_ids:
             known_object_ids = {}
```

Comment:
I know not related to this PR, but I saw this code the other day, why wouldn't this be set to `self.entities` and default to test against known entities ?

Reply:
Yes, that could probably work.